Sequence for .chezmoiexternal in Application Order

## What exactly are you trying to do?
Installing files using git-repo from .chezmoiexternal and it already mentions that we need git binary in the $PATH. So trying to figure out the sequence of .chezmoiexternal and run_ scripts

## What have you tried so far?
I want to know how .chezmoiexternal fits in this [application-order](https://www.chezmoi.io/reference/application-order/). So on a fresh Debian system if git is not installed my run_once_before installs git. But even though looks like .chezmoiexternal fails to find git.

## Where else have you checked for solutions?

* [X] I have read [chezmoi's user guide](https://chezmoi.io/user-guide/), and not found the answer.
* [X] I have searched [chezmoi's reference guide](https://chezmoi.io/reference/), and not found the answer.
* [ ] Other, please give details.

## Output of any commands you've tried with `--verbose` flag

So this is the snipped output from a GitHub Action log where I am testing my dotfiles in multiplatform. The run_ script is run with -x for debugging purpose.

```console
+ /usr/bin/apt -y install git

WARNING: apt does not have a stable CLI interface. Use with caution in scripts.

Reading package lists...
Building dependency tree...
Reading state information...
.
.
.
Setting up git (1:2.30.2-1) ...
Processing triggers for libc-bin (2.31-13+deb11u3) ...
+ which git
/usr/bin/git
.
.
.
chezmoi: exec: "git": executable file not found in $PATH
Error: Process completed with exit code 1.
```

This just fails when I am trying to use a git-repo in .chezmoiexternal

## Output of `chezmoi doctor`

## Additional context

I think once the installation script of chezmoi begins it fails to find new binaries installed in between run_ scripts and other actions.

--- a/pkg/chezmoi/lookpath.go
+++ b/pkg/chezmoi/lookpath.go
@@ -5,29 +5,25 @@ import (
 	"sync"
 )
 
-type lookPathCacheValue struct {
-	path string
-	err  error
-}
-
 var (
 	lookPathCacheMutex sync.Mutex
-	lookPathCache      = make(map[string]lookPathCacheValue)
+	lookPathCache      = make(map[string]string)
 )
 
-// LookPath is like os/exec.LookPath except that results are cached.
+// LookPath is like os/exec.LookPath except that the first positive result is
+// cached.
 func LookPath(file string) (string, error) {
 	lookPathCacheMutex.Lock()
 	defer lookPathCacheMutex.Unlock()
 
-	if result, ok := lookPathCache[file]; ok {
-		return result.path, result.err
+	if path, ok := lookPathCache[file]; ok {
+		return path, nil
 	}
 
 	path, err := exec.LookPath(file)
-	lookPathCache[file] = lookPathCacheValue{
-		path: path,
-		err:  err,
+	if err == nil {
+		lookPathCache[file] = path
 	}
+
 	return path, err
 }

